examples/block_causal_attention/block_causal_attention_varlen.py

```
@@ -0,0 +1,524 @@
+@tilelang.jit
+def _bwd_preprocess_varlen_template(heads: int, dim: int, dtype: str = "bfloat16"):
+    accum_dtype = "float32"
+    block_size = 64
+    # prep sweeps dim in 64-wide tiles; a non-multiple-of-64 tail would copy past the last column.
+    assert dim % block_size == 0, f"Delta preprocessing requires dim divisible by {block_size}, got {dim}"
+    total_tokens = T.dynamic("total_tokens")
+    shape = [total_tokens, heads, dim]
+
+    @T.prim_func
+    def prep(
+        O: T.Tensor(shape, dtype),
```

Ruff (0.15.20)

[error] Ambiguous variable name: `O`

(E741)